src/http/client.rs

```diff
@@ -262,6 +267,13 @@ impl EspHttpConnection {
         Ok(())
     }
 
+    pub fn flush_response(&mut self) -> Result<(), EspError> {
```

Comment:
if we call it everytime its needed in initate_request, is there any reason to have this methods still publicly available. E.g in what other scenario a user would need to call it?

Reply:
At first, I thought it'd be nice if the user could flush the response with `client.connection().flush_request()`. But then, like you, I thought: why use it if the response is going to be flushed anyway in `initiate_request`? Also, using `flush_request()` publicly could introduce issues unexpected issues. (Also, maybe we shouldn't introduce features that users aren't requesting.) 

So yeah, I think it's not worth making it public (I changed the code).